astropy/io/registry/core.py

```diff
@@ -362,16 +365,14 @@ def write(self, data, *args, format=None, **kwargs):
             .. versionadded:: 4.3
         """
         # Expand a tilde-prefixed path if present in args[0]
-        args = _expand_user_in_args(args)
+        if len(args):
+            args = (_expand_user_in_arg0(args[0]), *args[1:])
```

Comment:
I preferred when dirty details where hidden in this function...

Reply:
was it ? line 374 (old) was essentially the same thing and didn't belong in the function.
Besides, putting the `if len(args):` branching on the function makes it much harder to annotate clearly.